libs/os/os.go

```diff
@@ -13,19 +13,19 @@ type logger interface {
 	Info(msg string, keyvals ...interface{})
 }
 
-// TrapSignal catches the SIGTERM/SIGINT and executes cb function. After that it exits
-// with code 0.
+// TrapSignal catches SIGTERM and SIGINT, executes the cleanup function,
+// and exits with code 0.
 func TrapSignal(logger logger, cb func()) {
 	c := make(chan os.Signal, 1)
 	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+
 	go func() {
-		for sig := range c {
-			logger.Info(fmt.Sprintf("captured %v, exiting...", sig))
-			if cb != nil {
-				cb()
-			}
-			os.Exit(0)
+		sig := <-c
```

Comment:
I've removed the for loop as this is enough to block. Plus, decreasing indentation slightly improves readability.